### ISSUE TITLE
Custom Server: TypeError: Cannot add property webpackWatcher, object is not extensible

### ISSUE THREAD

**Issue by RasCarlito** [zeit/next.js#7998]
# Bug report

## Describe the bug

Trying to launch next.js using custom server API to integrate it into [strapi](https://strapi.io/) process fails with error :
```
TypeError: Cannot add property webpackWatcher, object is not extensible
    at resolve (/Users/carlogren/www/perso/next-strapi/node_modules/next/dist/server/next-dev-server.js:6:142)
```

This works fine with earlier versions of next.js (for example v8.1.0)

## To Reproduce

Clone following project https://github.com/RasCarlito/next-strapi

The next.js integration is developed [in a strapi middleware](https://github.com/RasCarlito/next-strapi/blob/master/middlewares/strapi-middleware-next/index.js).

Install dependencies with `npm i` then execute `npm run develop` to launch server. Checks terminal logs for errors. Access home page through http://localhost:1337/

Full log:
```
~/www/perso/next-strapi master 10s
❯ npm run develop

> next-integration@1.0.0 develop /Users/carlogren/www/perso/next-strapi
> strapi develop

> Using external babel configuration
> Location: "/Users/carlogren/www/perso/next-strapi/.babelrc"
[2019-07-16T14:22:15.027Z] info Time: Tue Jul 16 2019 16:22:15 GMT+0200 (Central European Summer Time)
[2019-07-16T14:22:15.028Z] info Launched in: 5897 ms
[2019-07-16T14:22:15.028Z] info Environment: development
[2019-07-16T14:22:15.028Z] info Process PID: 23285
[2019-07-16T14:22:15.028Z] info Version: 3.0.0-beta.9 (node v10.15.3)
[2019-07-16T14:22:15.029Z] info To shut down your server, press <CTRL> + C at any time

[2019-07-16T14:22:15.029Z] info ☄️  Admin panel: http://localhost:1337/admin
[2019-07-16T14:22:15.029Z] info ⚡️ Server: http://localhost:1337

[ ready ] compiled successfully
TypeError: Cannot add property webpackWatcher, object is not extensible
    at resolve (/Users/carlogren/www/perso/next-strapi/node_modules/next/dist/server/next-dev-server.js:6:142)
    at new Promise (<anonymous>)
    at DevServer.startWatcher (/Users/carlogren/www/perso/next-strapi/node_modules/next/dist/server/next-dev-server.js:5:468)
    at DevServer.prepare (/Users/carlogren/www/perso/next-strapi/node_modules/next/dist/server/next-dev-server.js:6:1241)
    at process._tickCallback (internal/process/next_tick.js:68:7)
[ wait ]  compiling ...
[ ready ] compiled successfully
```

Even though it says `compiled successfully`, as the `app.prepare()` promise rejected the `app.getRequestHandler()` does not seem to resolve to a working `handle` method.

Works fine when launching next.js on its own in project root:
```
npx next ./src
```

## Expected behavior

next.js should build without errors and then be accessible from http://localhost:1337/

## System information

- OS: macOS v10.32.2
- Version of Next.js: 9.0.1 (also tried with v9.0.2-canary.0)

**Comment by timneutkens** [comment]
Try adding a `"scripts"` entry for `next` to your package.json, I assume it's using a globally installed version of `next` 🤔

**Comment by RasCarlito** [comment]
Thanks for your reply @timneutkens 😃 

The error happens when using `next` custom server API [in this middleware](https://github.com/RasCarlito/next-strapi/blob/master/middlewares/strapi-middleware-next/index.js).

I don't have `next` installed globally. I just used `npx` to launch `next` on its own instead of launching it integrated into `strapi` process to demonstrate that there is a difference between `next` CLI and `next` custom server API.

**Comment by RasCarlito** [comment]
Ok so I tried the simple `custom-server-koa` example and it works fine on its own. My repository should actually work the same 🤔 

I'll continue digging but I guess it's more of an issue somewhere in my implementation of the custom server.

**Comment by RasCarlito** [comment]
Ok I found some workaround. I had to make sure that `app.prepare()` ran completely before I declared my `koa` router in the `strapi` middleware.

Why running `app.prepare()` at the same time as creating `koa` router ends up blocking the object onto which `next` tries to attach `webpackWatcher` I don't know.

Here is [the commit fixing the issue in my implementation](https://github.com/RasCarlito/next-strapi/commit/87135edfc9634921644f2df3b8b0d55a29604211). Now it actually behaves like the `custom-server-koa` example.

Closing this, sorry to have bothered you with this 😅
